RN: Add Icon Button Block Editor component.

enejb opened the pull request:
## Description
This PR adds the menu-item component into its own component now called "IconButton". 
The main reason for splitting up the component is to be able to reuse it in 3rd party blocks. 

## How has this been tested?
Since this a mobile only component for now it has been tested in the following way. 

It has been tested using the mobile demo app on both Android and iOS. 
1. After clicking the (+) the insert should appear as before. 
2. Select the column block. Notice that the layout column selector looks the same as before. 
3. Select one of the layouts. Notice that it works as before.

## Screenshots <!-- if applicable -->
Andorid
![Screen Shot 2020-09-09 at 4 46 50 PM](https://user-images.githubusercontent.com/115071/92665709-121ec000-f2bc-11ea-9ffb-c20fee1c4a51.png)
![Screen Shot 2020-09-09 at 4 46 32 PM](https://user-images.githubusercontent.com/115071/92665728-1ea31880-f2bc-11ea-945c-e5332caba605.png)

iOS
![Screen Shot 2020-09-09 at 4 45 46 PM](https://user-images.githubusercontent.com/115071/92665731-1fd44580-f2bc-11ea-874b-a1ada3087f9c.png)
![Screen Shot 2020-09-09 at 4 45 01 PM](https://user-images.githubusercontent.com/115071/92665735-206cdc00-f2bc-11ea-8e5c-6f5b1afbc8dc.png)

## Types of changes
This is mostly a refactor so that we can use the same use the new component in 3rd party plugins.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->
- [x] I've included developer documentation if appropriate. <!-- Handbook: https://developer.wordpress.org/block-editor/ -->
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. <!-- React Native mobile Gutenberg guidelines: https://github.com/WordPress/gutenberg/blob/master/docs/contributors/native-mobile.md -->

enejb committed:
RN: Add Icon Button Block Editor component.

enejb committed:
Add the missing styles

geriux reviewed (commented):
Hey @enejb 👋 

Thanks for moving this component into its own so it can be reused!

I think this component (that is mobile-only for now) should be in the `packages/components/src/mobile` folder.

By moving it there, it'll prevent [exposing](https://github.com/WordPress/gutenberg/blob/master/docs/tool/manifest.js#L11) the README file in the [Block Editor Handbook](https://developer.wordpress.org/block-editor/components/) docs.

What do you think?

enejb committed:
Add read me descriptions

enejb committed:
Move the IconButton component to components/mobile

enejb commented:
Thanks for the feedback @geriux 

I went and moved the component from `block-editor/components` to `components/mobile`. 

It turns out that there is already a component. Called `IconButton` that got deprecated in favor of just Button. 
I replaced the current IconButton from being included and replaced it with the new mobile/IconButton. 

Do we need to worry about deprecation on mobile? I wasn't able to find any (deprecated usage of) `IconButton` in the code base. 

Should we maybe rename the component to something else?

enejb committed:
Remove the duplicate styles that were found in styles.native.scss

geriux commented:
> I went and moved the component from `block-editor/components` to `components/mobile`.

Thank you for the changes @enejb!

> It turns out that there is already a component. Called `IconButton` that got deprecated in favor of just Button. 
> Do we need to worry about deprecation on mobile? I wasn't able to find any (deprecated usage of) `IconButton` in the code base. 

I checked the PR when it was [deprecated](packages/block-library/src/media-text/media-container.native.js) and looks like we were only using it in Media & Text block but it was changed in favor of the new `Button` so I think we shouldn't worry about the deprecation.

> Should we maybe rename the component to something else?

I think we should, it might cause confusion if other people start seeing references to it, what I don't know is what name could work for it =P

geriux commented on packages/components/src/mobile/icon-button/README.md:
```diff
@@ -0,0 +1,55 @@
+IconButton
+=============================
+
+Proved the IconButton style button for the mobile editor used in the inserter. 
```
Is "Proved" a typo? I'm trying to understand this phrase 😅

enejb replied:
Yes that is a typo.

enejb committed:
Update IconButton to InserterButton component

enejb commented:
@geriux I updated the component to be called InserterButton. I don't have a better name yet :( 

What do you think? Would that work?

geriux commented on packages/components/src/mobile/inserter-button/index.native.js:
```diff
@@ -16,6 +16,11 @@ import { __ } from '@wordpress/i18n';
  */
 import styles from './style.scss';
 
+export const inserterButtonStyles = {
```
What do you think if instead of exporting the styles separately we do something like [we do for the BottomSheet ](https://github.com/WordPress/gutenberg/blob/master/packages/components/src/mobile/bottom-sheet/index.native.js#L466-L478)?

So we have something like:

```
const InserterButton = withPreferredColorScheme( MenuItem );

InserterButton.Styles = inserterButtonStyles;

export default InserterButton;
```

enejb replied:
Yeah I like that pattern. Lets go with that! Thanks for pointing it out!

geriux commented on packages/components/src/mobile/inserter-button/README.md:
```diff
@@ -0,0 +1,55 @@
+InserterButton
+=============================
+
+InserterButton is a typo of button component. 
```
```suggestion
InserterButton is a type of button component. 
```

geriux reviewed (commented):
Hey @enejb just left two comments. I like the `InsertButton` name by the way 👍

enejb committed:
Fix typo

Co-authored-by: Gerardo Pacheco <gerardo.pacheco@automattic.com>

enejb committed:
Update the insert button style export

geriux reviewed (approved):
LGTM! Thanks for the changes @enejb!
